src/app.js

```diff
@@ -300,32 +300,24 @@ class Messenger extends EventEmitter {
       attachments
     } = message;
 
-    if (this.options.emitGreetings && this.greetings.test(text)) {
-      const firstName = session.profile && session.profile.first_name.trim() || '';
-      const surName = session.profile && session.profile.last_name.trim() || '';
-      const fullName = `${firstName} ${surName}`;
-
-      this.emit('text.greeting', new Response(this, {event, senderId, session, firstName, surName, fullName}));
-      return;
-    }
-
-    if (this.help.test(text)) {
-      this.emit('text.help', new Response(this, {event, senderId, session}));
+    if (this.emitOptionalEvents(event, senderId, session, text)) {
       return;
     }
 
     if (quickReply) {
-      debug('message.quickReply payload: "%s"', quickReply.payload);
+      const text = quickReply.payload;
+      const normalizedText = text.toLowerCase().trim();
+      debug('message.quickReply payload: "%s"', text);
 
-      this.emit('text', new Response(this, {event, senderId, session, source: 'quickReply', text: quickReply.payload}));
-      this.emit('message.quickReply', new Response(this, {event, senderId, session, payload: quickReply.payload}));
```

Comment:
Was having trouble following some of it, but IRL convo and we're good.

Conclusions:
* `message.postback` and `message.quickReply` should retain the `payload` attribute to avoid developer confusion
* after this PR and #55 land, we want to address documentation on why we went this route, as it results from our developer experience involving how to handle text users type that mimics the `title` of a postback or quick reply. I'll add this as a follow up task.